### ISSUE TITLE
would it be possible to expose the datetime of reactions

### ISSUE THREAD

**Issue by vesper8** [cybercog/laravel-love#192]
This is a continuation of an idea started in https://github.com/cybercog/laravel-love/issues/181

I would find it extremely useful to be able to easily retrieve the reaction date when using scopeWhereReactedTo or scopeWhereReactedBy

I have users liking other users and I would like to be able to display to a user who has liked them and when specifically that like has been received

Right now those scopes return the users that have reacted to/by the specfied reaction type.. but I'm not seeing an easy way to also retrieve more details about that reaction, namely the datetime it occurred at

The following lends itself to a n+1 problem.. but this is the workaround I came up with to retrieve the raw reaction from the db table in order to expose the datetime of the reaction itself which is useful in my case

```
$rawReaction = (new Love)->getRawReaction($user, User::find(request('viaResourceId')), 'LIKE');
```

```
<?php

namespace App\Helpers;

use App\Models\LoveReaction;

class Love
{
    public function getRawReaction($reactable, $reacterable, $reactionType)
    {
        $rawReaction = LoveReaction::where('reactant_id', $reactable->love_reactant_id)
        ->where('reacter_id', $reacterable->love_reacter_id)
        ->where('reaction_type_id', config('love.reaction-types')[$reactionType])
        ->first();

        return $rawReaction;
    }
}
```

```
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class LoveReaction extends Model
{
    protected $table = 'love_reactions';
}
```

**Comment by antonkomarev** [comment]
I will try to find time to look forward to it.

**Comment by antonkomarev** [comment]
@vesper8 could you try new scopes in master branch (before the release)? It looks like they already have required feature, or I'm missing something.

```php$usersReactedToMe = User::query()
    ->whereReactedTo($myUser)
    ->get();

dd($usersReactedToMe->first()->toArray());
```

Output:
```php
array:6 [
  "id" => 1
  "love_reactant_id" => 2
  "love_reacter_id" => 1
  "name" => "Pearlie Bergstrom"
  "created_at" => "2023-01-05T19:44:27.000000Z"
  "updated_at" => "2023-01-05T19:44:27.000000Z"
]
```

**Comment by antonkomarev** [comment]
This happens because Laravel using sub-queries:
```sql
select *
from "users"
where exists (
    select *
    from "love_reacters"
    where "users"."love_reacter_id" = "love_reacters"."id"
    and exists (
        select *
        from "love_reactions"
        where "love_reacters"."id" = "love_reactions"."reacter_id"
        and "reactant_id" = ?
    )
)
```

This kind of magic doesn't allow us to add reaction related data to outer select. It will be possible if we will start using joins, but it may lead us to total rewriting of the package. The more I look on it — more I think that such features should be done aside.

**Comment by vesper8** [comment]
Loving what I'm seeing @antonkomarev : )

While you're digging into this, it would be great to be able to query models that have recently reacted or been reacted to other models.. so using a timestamp when querying. So if I wanted to know the most "liked" model in the last 30 days for example.

`->whereReactedToBetween` or something like that : )

**Comment by antonkomarev** [comment]
Good idea @vesper8. The only thing I'm afraid of — bloating app models API with some magic scope methods.
